src/runtime_src/core/common/api/xrt_hw_context.cpp

```diff
@@ -278,6 +388,7 @@ post_alloc_hwctx(const std::shared_ptr<hw_context_impl>& handle)
   // called in XDP create a hw_context to the underlying implementation
   xrt_core::xdp::update_device(handle.get(), true);
   handle->get_usage_logger()->log_hw_ctx_info(handle.get());
+  handle->initialize_uc_log_buffer();
```

Comment:
We are sure initialize_uc_log_buffer() belongs in this function and not in constructor initializer list?  This function is specific to XDP meaning it is called only when XDP is configured specially in ini.

Reply:
Hi Soren, are you talking about post_alloc_hwctx function?
The post_alloc_hwctx is a common function (to avoid code duplicacy) called with hw ctx impl pointer created from different constructors.

Reply:
got it thanks, my oversight.